### PR TITLE
BOLT-11 Edits

### PR DESCRIPTION
**Pull request by shannona**
More line editing and clarifications, also includes typo-fix from @dimitris-t in #306 and table of contents from @bcongdon in #310.

### CODE REVIEW

File: 11-payment-encoding.md [hunk 1]
```diff
@@ -90,29 +107,29 @@ field specified below).
...
+* `n` (19): `data_length` 53. 33-byte public key of the payee node
+* `h` (23): `data_length` 52. 256-bit description of purpose of payment (SHA256). This is used to commit to an associated description that is too long to fit, such as on a web page.
+* `x` (6): `data_length` variable. `expiry` time in seconds (big-endian). Default is 3600 (1 hour) if not specified.
 * `c` (24): `data_length` variable. `min_final_cltv_expiry` to use for the last HTLC in the route. Default is 9 if not specified.
-* `f` (9): `data_length` variable, depending on version. Fallback on-chain address: for bitcoin, this starts with a 5 bit `version`; a witness program or P2PKH or P2SH address.
-* `r` (3): `data_length` variable.  One or more entries containing extra routing information for a private route; there may be more than one `r` field, too.
+* `f` (9): `data_length` variable, depending on version. Fallback on-chain address: for bitcoin, this starts with a 5-bit `version` and contains a witness program or P2PKH or P2SH address.
+* `r` (3): `data_length` variable. one or more entries containing extra routing information for a private route; there may be more than one `r` field
```
Comment (rustyrussell) [inline 11-payment-encoding.md]: lower-casing after the period seems weird; is that a thing?

**Comment by rustyrussell** [comment]
You will also need to add "bitcoins" to the wordlist in .aspell.en.pws.  It's case-sensitive.

File: 11-payment-encoding.md [hunk 2]
```diff
@@ -63,25 +80,25 @@ The `amount` is encoded into the human readable part, as it's fairly
...
 
 # Data Part
 
-The data part consists of multiple sections:
+The data part of a Lightning invoice consists of multiple sections:
 
 1. `timestamp`: seconds-since-1970 (35 bits, big-endian)
-1. Zero or more tagged parts.
-1. `signature`: bitcoin-style signature of above. (520 bits)
+1. Zero or more tagged parts
```
Comment (toadlyBroodle) [inline 11-payment-encoding.md]: zero...

File: 11-payment-encoding.md [hunk 3]
```diff
@@ -4,36 +4,53 @@ A simple, extensible QR-code-ready protocol for requesting payments
 over Lightning.
 
 # Table of Contents
+
+  * [Encoding Overview](#encoding-overview)
+    * [Requirements](#requirements)
+  * [Human-Readable Part](#human-readable-part)
+    * [Requirements](#requirements-1)
+    * [Rationale](#rationale)
```
Comment (toadlyBroodle) [inline 11-payment-encoding.md]: ToC sections for other BOLTs do not include 'Requirements' and 'Rationale' sections

File: 11-payment-encoding.md [hunk 4]
```diff
@@ -90,29 +107,29 @@ field specified below).
 
 ## Rationale
 
-`signature` covers an exact number of bytes because although the SHA-2
-standard actually supports hashing in bit boundaries, it's not widely
-implemented.  The recovery ID allows public key recovery, so the
+`signature` covers an exact number of bytes even though the SHA-2
+standard actually supports hashing in bit boundaries because it's not widely
```
Comment (toadlyBroodle) [inline 11-payment-encoding.md]: ...boundaries, because...

File: 11-payment-encoding.md [hunk 5]
```diff
@@ -90,29 +107,29 @@ field specified below).
...
-Currently defined Tagged Fields are:
+Currently defined tagged fields are:
 
-* `p` (1): `data_length` 52.  256-bit SHA256 payment_hash: preimage of this provides proof of payment.
-* `d` (13): `data_length` variable.  short description of purpose of payment (ASCII),  e.g. '1 cup of coffee'
-* `n` (19): `data_length` 53.  The 33-byte public key of the payee node.
-* `h` (23): `data_length` 52.  256-bit description of purpose of payment (SHA256).  This is used to commit to an associated description which is too long to fit, such as may be contained in a web page.
-* `x` (6): `data_length` variable.  `expiry` time in seconds (big-endian). Default is 3600 (1 hour) if not specified.
+* `p` (1): `data_length` 52. 256-bit SHA256 payment_hash. Preimage of this provides proof of payment
+* `d` (13): `data_length` variable. short description of purpose of payment (ASCII),  e.g. '1 cup of coffee'
```
Comment (toadlyBroodle) [inline 11-payment-encoding.md]: ...variable. Short...

File: 11-payment-encoding.md [hunk 6]
```diff
@@ -145,12 +162,12 @@ A writer MAY include one or more `f` fields. For bitcoin payments, a writer MUST
...
-public channel associated with its public key.  The `r` field MUST contain
+A writer MUST include at least one `r` field if there is not a
+public channel associated with its public key. The `r` field MUST contain
 one or more ordered entries, indicating the forward route from a
-public node to the final destination.  For each entry, the `pubkey` is the
-node ID of the start of the channel, `short_channel_id` is the short channel ID
-field to identify the channel, `fee_base_msat`, `fee_proportional_millionths` and `cltv_expiry_delta` are as specified in [BOLT #7](07-routing-gossip.md#the-channel_update-message).  A writer MAY include more than one `r` field to
+public node to the final destination. For each entry, the `pubkey` is the
+node ID of the start of the channel; `short_channel_id` is the short channel ID
+field to identify the channel; and `fee_base_msat`, `fee_proportional_millionths` and `cltv_expiry_delta` are as specified in [BOLT #7](07-routing-gossip.md#the-channel_update-message). A writer MAY include more than one `r` field to
```
Comment (toadlyBroodle) [inline 11-payment-encoding.md]: ...`fee_proportional_millionths`, and...

File: 11-payment-encoding.md [hunk 7]
```diff
@@ -285,7 +302,7 @@ Breakdown:
   * `xysxxatsyp3k7enxv4js`: '1 cup coffee'
 * `x`: expiry time
   * `qz`: `data_length` (`q` = 0, `z` = 2. 0 * 32 + 2 == 2)
-  * `pu`: 60 seconds (`p` = 1, `u` = 28.  1 * 32 + 28 == 60)
+  * `pu`: 60 seconds (`p` = 1, `u` = 28. 1 * 32 + 28 == 60)
```
Comment (toadlyBroodle) [inline 11-payment-encoding.md]: ...28; 1... use semicolon or colon instead of period?

File: 11-payment-encoding.md [hunk 8]
```diff
@@ -354,8 +371,8 @@ Breakdown:
   * `3` = 17, so P2PKH address
   * `qjmp7lwpagxun9pygexvgpjdc4jdj85f`: 160 bit P2PKH address
 * `r`: tagged field: route information
-  * `9y`: `data_length` (`9` = 5, `y` = 4.  5 * 32 + 4 = 164)
-    `q20q82gphp2nflc7jtzrcazrra7wwgzxqc8u7754cdlpfrmccae92qgzqvzq2ps8pqqqqqqqqqqqq9qqqvpeuqafqxu92d8lr6fvg0r5gv0heeeqgcrqlnm6jhphu9y00rrhy4grqszsvpcgpy9qqqqqqqqqqqq7qqzq`: pubkey `029e03a901b85534ff1e92c43c74431f7ce72046060fcf7a95c37e148f78c77255`, `short_channel_id` 0102030405060708, `fee_base_msat` 1 millisatoshi, `fee_proportional_millionths` 20, `cltv_expiry_delta` 3.  pubkey `039e03a901b85534ff1e92c43c74431f7ce72046060fcf7a95c37e148f78c77255`, `short_channel_id` 030405060708090a, `fee_base_msat` 2 millisatoshi, `fee_proportional_millionths` 30, `cltv_expiry_delta` 4.
+  * `9y`: `data_length` (`9` = 5, `y` = 4. 5 * 32 + 4 = 164)
```
Comment (toadlyBroodle) [inline 11-payment-encoding.md]: use semicolon or colon seperator instead of period?

File: 11-payment-encoding.md (hunk 1 above)
Comment (shannona) [reply 11-payment-encoding.md]: No, probably not.

File: 11-payment-encoding.md (hunk 3 above)
Comment (shannona) [reply 11-payment-encoding.md]: Thanks.

File: 11-payment-encoding.md (hunk 2 above)
Comment (shannona) [reply 11-payment-encoding.md]: Done.

File: 11-payment-encoding.md (hunk 4 above)
Comment (shannona) [reply 11-payment-encoding.md]: Done.

File: 11-payment-encoding.md (hunk 6 above)
Comment (shannona) [reply 11-payment-encoding.md]: Done.

File: 11-payment-encoding.md (hunk 7 above)
Comment (shannona) [reply 11-payment-encoding.md]: think you're right; that's more readable. Changing throughout.

**Comment by shannona** [comment]
Review fixes should now all be in place now. @toadlyBroodle I went ahead and adjusted all the equations, as you suggested.

**Pull request merged**